src/liballoc/vec.rs

```diff
@@ -636,6 +638,7 @@ impl<T> Vec<T> {
     /// assert_eq!(slice.into_vec().capacity(), 3);
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[inline(always)]
```

Comment:
OTOH, removing this one makes the `into_boxed_slice` calls not be optimised anymore when the function is duplicated. Even just `#[inline]` makes rustc not inline the calls anymore.

Reply:
Hm so in general `#[inline(always)]` is an antipattern as it can cause compile time to baloon very quickly. Mind if I poke around a bit at the examples you were looking at to see what's going on?

Reply:
Sure!

```rust
#![crate_type = "lib"]

#[no_mangle]
pub fn foo() -> Box<[u8]> {
    vec![0].into_boxed_slice()
}

#[no_mangle]
pub fn bar() -> Box<[u8]> {
    vec![0].into_boxed_slice()
}
```

Reply:
Hm ok I'm tempted though to leave this as-is and let LLVM decide these sorts of things, we've had bad experiences in the past optimizing too much for microbenchmarks unfortunately

Reply:
Oh just this one particular instance of the attribute. The function is generic so it's already candidate to be inlined everywhere, and it sounds like `#[inline]` doesn't otherwise work here so `#[inline(always)]` is going directly against LLVM's heuristics, which has caused a lot of problems for us historically